tweakwcs/wcsimage.py

```diff
@@ -1678,6 +1702,11 @@ def cross(o, a, b):
 
     # convert x, y vertex coordinates to RA & DEC:
     ra, dec = wcs(ptx, pty)
+
+    # TODO: for strange reasons, occasionally ra[0] != ra[-1] and/or
+    #       dec[0] != dec[-1] (even though we close the polygon in the
+    #       previous two lines). Then SphericalPolygon fails because
+    #       points are not closed. Threfore we force it to be closed:
     ra[-1] = ra[0]
```

Comment:
Do you still want to simply replace the last result with the first entry in the output list?  This assumes that the 'ra[-1]' is essentially floating-point equivalent to `ra[0]` already.  Is this assumption ALWAYS correct, as it is not obvious from the code alone (at least to me)?  If this assumption can not be guaranteed to ALWAYS be correct, then this line is introducing a potentially major change to the shape of the output polygon.